compiler/src/iree/compiler/DispatchCreation/SetSplitReductionSizes.cpp

```diff
@@ -398,6 +398,17 @@ struct SetSplitReductionSizesPass final
     // The constants below are determined based on empirical data.
     const int64_t ratioThreshold = 384;
     const int64_t largeKSize = 24576;
+    const int64_t largeMNSize = 1024;
```

Comment:
This constant looks fairly aggressive to me: each MFMA only can handle mxn of 16x16 on mi300x. By setting this limit to 1024, it will allow 64 MFMAs to consume the entire M/N. I think we should limit it further to 1 workgroup tile, say 256. This allows 1 tile per parallel dimension to still not go to split K (with decent tile size). 

BTW, this is purely based on my intuition. So if you find a counter example, let me know.